### DISCUSSION TITLE
How to Read/Understand Code Logically

### DISCUSSION THREAD

**Discussion by Ziv08** [Cyfrin/foundry-full-course-cu#2378] [Q&A]
Hi guys, it's my first post! 
I'm at the second learning section on Cyfrin Updraft. It's also my first time to learn computer programming, freshly new beginner in coding. My question is how to read code, or having a logical coding mindset? I can read in English even though it's not my first language. I just can't understand it fully while I code along with lessons. I appreciate any comments and helps! BTW the course are really awsome @PatrickAlphaC

**Comment by EngrPips** [comment]
Hello @Ziv08, I know you expect some tips and tricks like telling you to invert your system and read the code upside down, and the code will gradually sink into your brain, and boom you will understand the code but honestly the only truth I know is that it will make more sense with time by the time you are six months into programming you will better than yourself in month one considering you are committed to it. so keep doing it and it will become more easy.

**Reply by Ziv08** [reply]
Thank you for the advice, I’ll keep doing it :)

**Reply by EngrPips** [reply]
Anytime.

**Comment by PhilippeOurselin** [comment]
Hello,

A tip I hope you will find useful: ***Reformulate the code you are reading in ordinary sentences.***

#### Example:

```solidity
function addNumbers(uint256 a, uint256 b) public returns(uint256) {}
```
Read this like that: *Let's make a new public function named `addNumbers` which takes two numbers as arguments, `a` and `b`, and which will return their sum.*

Hope that helps!

**Comment by EngrPips** [comment]
Interesting.
